sdks/python/apache_beam/io/gcp/dicomio_integration_test.py

```diff
@@ -0,0 +1,215 @@
+    # Search and compare the metadata of a persistent DICOM store.
+    input_dict = {}
+    input_dict['project_id'] = self.project
+    input_dict['region'] = REGION
+    input_dict['dataset_id'] = DATA_SET_ID
+    input_dict['dicom_store_id'] = PERSISTENT_DICOM_STORE_NAME
+    input_dict['search_type'] = "instances"
+
+    expected_dict = {}
+    expected_dict['result'] = self.expected_output_all_metadata
+    expected_dict['status'] = 200
+    expected_dict['input'] = input_dict
+    expected_dict['success'] = True
+
+    with self.test_pipeline as p:
+      results = (p | beam.Create([input_dict]) | DicomSearch())
+      assert_that(results, equal_to([expected_dict]), label='all search assert')
+
+  @attr('IT')
+  def test_dicom_search_refined_instances(self):
+    # Refine search and compare the metadata of a persistent DICOM store.
+    input_dict = {}
+    input_dict['project_id'] = self.project
+    input_dict['region'] = REGION
+    input_dict['dataset_id'] = DATA_SET_ID
+    input_dict['dicom_store_id'] = PERSISTENT_DICOM_STORE_NAME
+    input_dict['search_type'] = "instances"
+    input_dict['params'] = {
+        'StudyInstanceUID': 'study_000000001', 'limit': 500, 'offset': 0
+    }
+
+    expected_dict = {}
+    expected_dict['result'] = self.expected_output_refined_metadata
+    expected_dict['status'] = 200
+    expected_dict['input'] = input_dict
+    expected_dict['success'] = True
+
+    with self.test_pipeline as p:
+      results = (p | beam.Create([input_dict]) | DicomSearch())
+      assert_that(
+          results, equal_to([expected_dict]), label='refine search assert')
+
+  @attr('IT')
+  def test_dicom_store_instance_from_gcs(self):
+    # Store DICOM files to a empty DICOM store from a GCS bucket,
+    # then check if the store metadata match.
+    input_dict_store = {}
+    input_dict_store['project_id'] = self.project
+    input_dict_store['region'] = REGION
+    input_dict_store['dataset_id'] = DATA_SET_ID
+    input_dict_store['dicom_store_id'] = self.temp_dicom_store
+
+    expected_output = [True] * NUM_INSTANCE
+
+    with self.test_pipeline as p:
+      gcs_path = TEMP_FILES_PATH + "/dicom_files/*"
+      results = (
+          p
+          | fileio.MatchFiles(gcs_path)
+          | fileio.ReadMatches()
+          | UploadToDicomStore(input_dict_store, 'fileio')
+          | beam.Map(lambda x: x['success']))
+      assert_that(
+          results, equal_to(expected_output), label='store first assert')
+
+    input_dict_search = {}
+    input_dict_search['project_id'] = self.project
+    input_dict_search['region'] = REGION
+    input_dict_search['dataset_id'] = DATA_SET_ID
+    input_dict_search['dicom_store_id'] = self.temp_dicom_store
+    input_dict_search['search_type'] = "instances"
+
+    expected_dict = {}
+    expected_dict['result'] = self.expected_output_all_metadata
+    expected_dict['status'] = 200
+    expected_dict['input'] = input_dict_search
+    expected_dict['success'] = True
+
```

Comment:
in this test case you're running two pipelines. This will therefore take 10 minutes to run. Perhaps it's worth making this the single test case that tests upload and search (or perhaps remove the search pipeline from this case to avoid the super long runtime?)

Reply:
Pipeline reduced